state/model.go

```diff
@@ -558,6 +562,37 @@ func (m *Model) ModelTag() names.ModelTag {
 	return names.NewModelTag(m.doc.UUID)
 }
 
+// SetPassword sets the password for the model's agent.
+// TODO(caas) - consider a separate CAAS application entity
```

Comment:
this application entity comment can be removed

however, there is a separate CAASModel struct we should add these methods to instead